### ISSUE TITLE
[big Graph training]-about 1B nodes and  10B edges

### ISSUE THREAD

**Issue by lixusign** [dmlc/dgl#1819]
I’m sorry 。
I want to doing a transductive learning ,  like  deepwalk or line alg 
so,  I have a so big graph，maybe 1B:nodes， 10B:edges

i want to know ，in pytorch framework and dgl0.4 ，how mary memory I need？
and how did dgl work about  such a so big embedding_lookup_table storage and update 。

if i used tensorflow in another graphlib ,  i need more than 50 param servers to store and sharding a big tensor like shape [1B， dim] ,type=float32.

so i very puzzled about this.

is any possible I training this big graph in a single machine and use 500GB mem

**Comment by classicsong** [comment]
With 128 dim size, 1B nodes takes 512GB mem.
You can try our DeepWalk implementation with multigpu support, but I think you may need 600GB+ more memory.

Another solution is to use DGL-KE, which support distributed training for algorithms like transE, DistMult, etc.

**Comment by lixusign** [comment]
enen,  I  mean ，so big lookup table store in transductive learning algs cost memory ：

1B * 128 * 4 / 1024 /1024 /1024 = 4768GB

I dont understand , How pytorch  store and update this ?

**Comment by lixusign** [comment]
sorry I figure out with error  ， you are right about 600GB
